### ISSUE TITLE
RxJava Maven dependency scope=runtime but is required for compilation

### ISSUE THREAD

**Issue by patrickconant** [Netflix/Hystrix#730]
I've encountered this issue, but don't know enough about Gradle to help diagnose the issue. In versions of Hystrix prior to 1.4, the Maven dependency on io.reactivex:rxjava had scope=compile ([see pom.xml in Maven Central](https://repo1.maven.org/maven2/com/netflix/hystrix/hystrix-core/1.3.20/hystrix-core-1.3.20.pom)). With version 1.4 and later, the Maven dependency scope is set to "runtime" ([pom.xml in Maven Central](https://repo1.maven.org/maven2/com/netflix/hystrix/hystrix-core/1.4.0/hystrix-core-1.4.0.pom)). This causes Maven-based java compilation failures of projects that depend on Hystrix 1.4 or greater. In my case, the compilation error message is 'cannot access rx.Observable  class file for rx.Observable not found'.

It doesn't appear that anything has changed in the hystrix-core/build.gradle file -- the dependency looks right ("compile 'io.reactivex:rxjava:1.0.7'"). But something appears to have changed in how Gradle translates that into the maven pom.xml that's published to Maven Central.

There is a work-around for users running into this issue -- one can simply add a compile-time dependency on rxjava to their own project. That makes using Hystrix a bit more complicated and isn't standard handling for library dependencies.

**Comment by mattrjacobs** [comment]
Thanks for the report @patrickconant .  This was not an intentional change - I'll have to track down what's causing this - not sure off the top of my head.

**Comment by mattrjacobs** [comment]
@patrickconant / @axiak Looks like the switch to Nebula publishing broke this piece of the pom.  This bug exists in 1.4.0/1.4.1/1.4.2.

Also see https://github.com/ReactiveX/RxNetty/issues/292 for another example of this.

I've notified the Nebula owners and they have already got this in their list of fixes coming shortly.

As soon as that occurs, I'll do a release to pick up that change.

Until then, the only workaround is to explicitly depend on all Hystrix dependencies yourself :(

For hystrix-core, this should be archauis-core + rxjava + slf4j-api

**Comment by axiak** [comment]
Thanks for looking into it.. looks like we're going to ship a shaded version internally :(

**Comment by axiak** [comment]
FYI this is what we're using to ship our own corrected version of 1.4.2: https://git.hubteam.com/gist/maxiak/e88ec3c5997d3eabf374

**Comment by mattrjacobs** [comment]
FWIW, here's the nebula issue: https://github.com/nebula-plugins/nebula-publishing-plugin/issues/31

**Comment by spencergibb** [comment]
why would you need to publish your own version?  Don't you just need a pom with the right dependencies?  That's what we're doing for [spring-cloud-netflix](https://github.com/spring-cloud/spring-cloud-netflix).

**Comment by axiak** [comment]
@spencergibb it depends on who you're solving for. we have hundreds of projects that depend on hystrix that I'm trying to upgrade to a single new version.

**Comment by juristr** [comment]
For others, I got it to work with this Maven configuration:

```xml
<dependency>
	<groupId>com.netflix.hystrix</groupId>
	<artifactId>hystrix-core</artifactId>
	<version>1.4.3</version>
</dependency>
<dependency>
	<groupId>com.netflix.archaius</groupId>
	<artifactId>archaius-core</artifactId>
	<version>0.4.1</version>
</dependency>
<dependency>
	<groupId>io.reactivex</groupId>
	<artifactId>rxjava</artifactId>
	<version>1.0.7</version>
	<scope>compile</scope>
</dependency>
<dependency>
	<groupId>org.slf4j</groupId>
	<artifactId>slf4j-api</artifactId>
	<version>1.7.0</version>
</dependency>
```

**Comment by mhuisman** [comment]
This issue persists in version 1.4.7, fix above still applies

**Comment by taringamberini** [comment]
This issue persists in version 1.4.10.

As previously suggested by @juristr excplicitly declare `compile` scope in order to get your project compiled:

    <dependency>
        <groupId>io.reactivex</groupId>
        <artifactId>rxjava</artifactId>
        <version>${rxjava_version_requested_by_hystrix-core_x_y_z}</version>
        <scope>compile</scope>
    </dependency>

**Comment by jmaasing** [comment]
This issue persists in version 1.4.14

**Comment by chanwookpark** [comment]
Hi. Is this issue patched? 
I meet this error 'cannot access rx.Observable. class file for rx.Observable not found"

thanks you.

**Comment by aminjava** [comment]
Adding this dependency fixed the build for me 
	<dependency>
		<groupId>io.reactivex</groupId>
		<artifactId>rxjava</artifactId>
		<version>1.0.8</version>
	</dependency>

**Comment by mattrjacobs** [comment]
This should be fixed with #944 and available in the next release
